Lib/asyncio/taskgroups.py

```diff
@@ -177,6 +180,10 @@ async def _aexit(self, et, exc):
             finally:
                 exc = None
 
+        # If we wanted to raise an error, it would have been done explicitly
+        # above.  Otherwise, either there is no error or we want to suppress
+        # the original error.
+        return True
```

Comment:
does this bugfix need backporting to 3.12?

Reply:
Can you think of a case where this bug is visible to the user?  If it's visible, yes I'd make a separate PR with corresponding test that can be backported.

Reply:
What is the use-case for this code?
If the bug is present -- let's create a separate issue and make a fix with the backport.
Anyway, I don't see how is this change related to adding `.cancel()` method.

Reply:
Suppressing exceptions out of the context manager is certainly needed to implement `TaskGroup.cancel()`.  Without it, the following basic test will fail:

```python
async def test_taskgroup_cancel_body(self):
    count = 0
    async with asyncio.TaskGroup() as tg:
        tg.cancel()
        count += 1
        await asyncio.sleep(0)  # <-- CancelledError will leak out of context manager
        count += 1
    self.assertEqual(count, 1)
```

Note that the change isn't a blanket suppression.  Code prior to this `return True` explicitly raises any exception it wants propagated out of the context manager.

Reply:
Ah it's visible in the Traceback returned for external or "native" cancellation. Ie a cancellation that propagates out of run